rubrics/impl/src/main/java/org/sakaiproject/rubrics/impl/repository/AssociationRepositoryImpl.java

```diff
@@ -97,24 +94,27 @@ public List<ToolItemRubricAssociation> findByItemIdPrefix(String toolId, String
         return session.createQuery(query).list();
     }
 
-    public ToolItemRubricAssociation save(ToolItemRubricAssociation assoc) {
-
-        Rubric rubric = assoc.getRubric();
-        List<ToolItemRubricAssociation> associations = rubric.getAssociations();
-        int index = associations.indexOf(assoc);
-
-        if (index != -1) {
-            associations.set(index, assoc);
-        } else {
-            associations.add(assoc);
-        }
+    public ToolItemRubricAssociation save(ToolItemRubricAssociation association) {
+        ToolItemRubricAssociation mergedAssociation = null;
 
-        rubric = rubricRepository.save(rubric);
-        if (index != -1) {
-            return rubric.getAssociations().get(index);
+        Long associationId = association.getId();
+        if (associationId != null) {
+            // updating an existing association we can merge it and return it as nothing changes on the rubric
+            mergedAssociation = (ToolItemRubricAssociation) sessionFactory.getCurrentSession().merge(association);
```

Comment:
I'm not sure I agree with this, tbh. This means that the state isn't consistent until the rubric is loaded  from the db once more, so if anybody in the future uses the rubric's list of associations, they may have a stale association in that list. Isn't it cleaner to just update it in the rubric and let the cascade handle the merge?

Reply:
Note that the association was simply updated hibernate will internally update it's references to the updated association so no harm there and no the list of associations would not be stale. We absolutely could let the merge on the rubric cascade the update to the association this is simply faster and it achieves the same goal. Now if it were adjusting the actual list of associations then you would take the approach that you mention which it does, in the subsequent lines after this.

Reply:
You are correct though in noticing this shortcut.